pkg/api/types.go

```diff
@@ -396,6 +398,24 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: hosts are Glusterfs hosts address array
+	Hosts []string `json:"hosts"`
```

Comment:
@rootfs: +1 on @erictune's thought on not requiring hosts in the API. What about using a kubernetes service or an endpoints object?